subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -112,10 +112,230 @@ int net_init_clock_via_sntp(void)
 }
 
 #ifdef CONFIG_NET_CONFIG_SNTP_INIT_RESYNC
+static void sntp_async_service_handler(struct net_socket_service_event *pev);
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(service_sntp_async, sntp_async_service_handler, 1);
+
+static void sntp_async_timeout(struct k_work *work);
+static K_WORK_DELAYABLE_DEFINE(sntp_async_timeout_work, sntp_async_timeout);
+
+static struct sntp_ctx sntp_async_ctx;
+static struct net_sockaddr_storage sntp_addr;
+static net_socklen_t sntp_addrlen;
+
+#define RESYNC_FAILED_INTERVAL K_SECONDS(CONFIG_NET_CONFIG_SNTP_INIT_RESYNC_ON_FAILURE_INTERVAL)
+
+static void sntp_async_timeout(struct k_work *work)
```

Comment:
Nit: AFAIK you can use `__unused` instead of `ARG_UNUSED()`
```suggestion
static void sntp_async_timeout(struct k_work __unused *work)
```

Reply:
I didn't see a single instance in our codebase where `__unused` was used with function argument, I'll stick to `ARG_UNUSED()` which is a common pattern.